src/Form/RegistrationFormType.php

```diff
@@ -13,25 +13,57 @@
 use Symfony\Component\Form\FormBuilderInterface;
 use Symfony\Component\OptionsResolver\OptionsResolver;
 use Symfony\Component\Validator\Constraints\IsTrue;
+use Symfony\Component\Validator\Constraints\NotBlank;
+use Symfony\Component\Validator\Constraints as Assert;
 
 class RegistrationFormType extends AbstractType
 {
     public function buildForm(FormBuilderInterface $builder, array $options): void
     {
         $builder
-            ->add('firstname', TextType::class)
-            ->add('lastname', TextType::class)
+            ->add('firstname', TextType::class, [
+                'constraints' => [
+                    new NotBlank([
+                        'message' => 'Vous devez renseigner un prénom.'
+                    ])]
+            ])
+            ->add('lastname', TextType::class, [
+                'constraints' => [
+                    new NotBlank([
+                        'message' => 'Vous devez renseigner un nom.'
+                    ])]
+            ])
             ->add('gender', ChoiceType::class, [
                 'choices' => [
                     'Féminin' => 'female',
                     'Masculin' => 'male',
                     'Non binaire' => 'non binary',
                     'Ne souhaite pas se prononcer' => null,
+                ],
+                'constraints' => [
+                    new NotBlank([
+                        'message' => 'Merci de renseigner un genre',
+                    ])
```

Comment:
the constraint works but it is also put in place when the person does not wish to disclose his gender